.github/workflows/codeql.yml

```diff
@@ -0,0 +1,53 @@
+name: CodeQL
+
+on:
+  push:
+    branches: [main, future/*, release/*]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: 29 22 * * 2
```

Comment:
This will let us catch issues early on PRs, however we also need to run CodeQL at least once every 30 days on our `main` branch to remain compliant (e.g. if there are no PRs for >30days).